.travis.yml

```diff
@@ -203,7 +203,19 @@ script:
         make check-integration || exit 1
         make check-themes || exit 1
         do_codecov functionaltests
-        do_codecov_gcov c_code
+
+        # Run gcov.  Upload for codecov directly, gets done later for coveralls.
+        # NOTE: files for the main target have the build dir as prefix, but
+        #       otherwise it is the source directory?!
```

Comment:
I'm not really sure what this comment means. What is "otherwise"? Which files is this about?

According to https://gcc.gnu.org/onlinedocs/gcc-4.7.3/gcc/Gcov-Data-Files.html#Gcov-Data-Files, the same directory as the object file is used, so build dir. However, I guess that if the compiler is invoked with a relative path, then the program ends up using a relative path as well...? I'm not really sure since I don't fully grasp what the "symptoms" are.

On an unrelated note: You recently opened quite a number of PRs and it felt to me that some of them were still WIP, so I ignored them. If you feel like some PR is "finally done", I would suggest to add a comment saying so, so that everyone knows about it. I'm mostly working based on GitHub notifications and so for me PRs can easily "fall through the cracks".